docs/developer/pull-request.md

```diff
@@ -0,0 +1,107 @@
+- Do your best to review in a timely manner. However, code reviews are time-consuming, maximize their benefit by
+  focusing on what’s highest value. This code review pyramid outlines a reasonable shorthand:
+  ![Code Review Pyramid](code-review-pyramid.jpeg)
+- Always review for: design, API semantics, [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and
+  maintainability practices, bugs and quality, efficiency and correctness.
+- Code review checklist:
+    - Do any changes need to made in documentation?
+        - Do public docs need updating?
+        - Have internal workflow and informational docs been impacted?
+    - Are there unit tests for the change?
+        - If the change is a bug fix, has a unit test been added or modified that catches the bug? All bug fixes should
+          be reproduced with a unit test before submitting any code. Then the code should be fixed so that the unit test
+          passes.
+        - If the change is a feature or new functionality. The expectation is to include a reasonable set of unit tests
+          which exercise positive and negative cases.
+    - Is this change backwards compatible? Is it causing breaking behavior?
+    - Are there any [Comment Tags](#comment-tags) in the change?
+
+## Commit Message Format
+
+The commit message for your final commit should use a consistent format:
+
+```text
+<One line summary of the change>
+
+Problem: Brief description of the problem. The “why”.
+
+Solution: Detailed description of the solution. Try to refrain from pseudo-code. Describe some of the analysis, thought
+process, and outline what was done. Think of a newcomer coming across this two years hence.
+
+Testing (optional): Description of tests that were run to exercise the solutions (unit tests, system tests,
+reproductions, etc.)
+
+```
+
+Here's an example:
+
+```text
+Make event stream handle large events
+
+Problem: The event stream would choke on events larger than 1K. This caused events to be silently dropped, and affected
+all downstream consumers.
+
+Solution: Removed upper limit on event size. Also, added an error handler to warn when events are being dropped for any
+reason.
+
+Added unit tests for events up to 1MB.
+```
+
+> **Note**
+> Do not put any customer identifying information in message. Say “a customer found…”, NOT “ACME Corp found…”. If customer generated data is included it must be redacted for any PII (or identifying information). This includes IPs, ports, names of applications and objects, and URL paths. If customers volunteered this information, do not propagate it externally.
+
+For additional help on writing good commit messages, see this [article](https://cbea.ms/git-commit/).
+
+# Comment Tags
+
+Common comment tags are FIXMEs and TODOs.
+
+A TODO is a developer note-to-self. TODOs _MUST_ be fixed before merging.
+
+A FIXME is for things that should be fixed in the future. FIXMEs can be merged, but they _MUST_ include your username and
```

Comment:
shall we ask not to create an issue before PR is reviewed? Because during a review, the reviewer might ask to resolve FIXME or consider a different approach?

Reply:
I'll rephrase to make that clear.